pkg/multicast/multicast.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"golang.org/x/net/ipv6"
+)
+
+var (
+	// v6Socket is the udp socket used to join/leave a multicast group
+	v6Socket net.PacketConn
+
+	mutex lock.Mutex
+
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "multicast")
+)
+
+// Pre-Defined Multicast Addresses
+// as defined in https://tools.ietf.org/html/rfc4291#section-2.7.1
+
+var (
+	// AllNodesIfcLocalMaddr is the multicast address that identifies the group of
+	// all IPv6 nodes, within scope 1 (interface-local)
+	AllNodesIfcLocalMaddr net.IP = []byte{
+		0xff, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1,
+	}
```

Comment:
Would that make sense to use something like this instead (here and below)? I think it would be more readable:
```suggestion
	AllNodesIfcLocalMaddr net.IP = net.ParseIP("ff01::1")
```